task/sql-tasks.js

```diff
@@ -67,18 +83,36 @@ async function task_1_3(db) {
  *
  */
 async function task_1_4(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT 
+        CustomerID AS "Customer Id",
+        COUNT(CustomerID) AS "Total number of Orders",
+        ROUND(COUNT(OrderID)*100/SUM(count(OrderID)) OVER(),5) AS "% of all orders"
+    FROM Orders
+    GROUP BY CustomerID
+    ORDER BY \`% of all orders\` DESC, CustomerID;
+    `);
+    return result[0];
 }
 
 /**
- * Return all products where product name starts with 'A', 'B', .... 'F' ordered by name.
+ * Return all products where product name starts with 'A', 'Bs', .... 'F' ordered by name.
  * | ProductId | ProductName | QuantityPerUnit |
  *
  * @return {array}
  *
  */
 async function task_1_5(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT 
+        ProductID AS "ProductId",
+        ProductName AS "ProductName",
+        QuantityPerUnit AS "QuantityPerUnit"
+    FROM Products
+    WHERE ProductName REGEXP '^[A-F]'

```

Comment:
That's possible to use simple `like` operator here. That's important because of indexes. Regular expression lookups can't use indexes

Reply:
I am not sure, but MySql doesn't support range notation( LIKE '[A-F]%' or '^[A-F]'). We can use only RLIKE or REGEX. Documentations - https://dev.mysql.com/doc/refman/5.6/en/string-comparison-functions.html#operator_like, StackOferflow - https://stackoverflow.com/questions/12612580/mysql-like-with-range-doesnt-work.

Reply:
> MySql doesn't support range notation( LIKE '[A-F]%' or '^[A-F]')

Yes, I know. "ABCDEF" are just 6 cases. So in case of bad performance it's better to use just list of like operators via OR.

Reply:
Ok